common/scripts/upgrade_docker.sh

```diff
@@ -25,161 +42,124 @@ __remove_services() {
-
-      __copy_script_remote "$host" "$install_worker_script" "$SCRIPTS_DIR_REMOTE"
-      __exec_cmd_remote "$host" "$worker_install_cmd"
-    done
-  fi
+    if [ $host == $ADMIRAL_IP ]; then
+      continue
+    fi
+
+    local script_name="installWorker.sh"
+    local install_worker_script="$SCRIPTS_DIR/$ARCHITECTURE/$OPERATING_SYSTEM/remote/$script_name"
+    local worker_install_cmd="WORKER_HOST=$host \
+      WORKER_JOIN_TOKEN=$SWARM_WORKER_JOIN_TOKEN \
+      WORKER_PORT=$port \
+      MASTER_HOST=$master_address \
+      RELEASE=$RELEASE \
+      NO_VERIFY_SSL=$NO_VERIFY_SSL \
+      ARCHITECTURE=$ARCHITECTURE \
+      OPERATING_SYSTEM=$OPERATING_SYSTEM \
+      INSTALLED_DOCKER_VERSION=$DOCKER_VERSION \
+      SHIPPABLE_HTTP_PROXY=$SHIPPABLE_HTTP_PROXY \
+      SHIPPABLE_HTTPS_PROXY=$SHIPPABLE_HTTPS_PROXY \
+      SHIPPABLE_NO_PROXY=$SHIPPABLE_NO_PROXY \
+      $SCRIPTS_DIR_REMOTE/$script_name"
+
+    __copy_script_remote "$host" "$install_worker_script" "$SCRIPTS_DIR_REMOTE"
+    __exec_cmd_remote "$host" "$worker_install_cmd"
+  done
 }
 
-__upgrade_docker_version() {
-  if $DOCKER_UPGRADE_REQUIRED; then
-    __process_msg "Upgrading docker to $DOCKER_VERSION"
+__update_docker() {
+  ## Note: this always needs to execute since it'll update proxy settings on
+  ## swarm master
 
-    ./$INSTALL_DOCKER_SCRIPT
-    rm -f $INSTALL_DOCKER_SCRIPT
-  fi
+  __process_msg "Upgrading docker to $DOCKER_VERSION"
+  __create_install_docker_script
+  ./$INSTALL_DOCKER_SCRIPT
+  rm -f $INSTALL_DOCKER_SCRIPT
 }
 
 __upgrade_awscli_version() {
+  __process_msg "Upgrading AWS cli to : $AWSCLI_VERSION"
   pip install awscli==$AWSCLI_VERSION
 }
 
 __restart_db_container() {
-  if $DOCKER_UPGRADE_REQUIRED; then
-    local db_container_name="db"
-    local db_container=$(sudo docker ps -a --format "{{.Names}}" | grep -w "db") || true
-    if [ ! -z "$db_container" ]; then
-      __process_msg "Found a stopped $db_container_name container, starting it"
-      sudo docker start "$db_container_name"
-      sleep 3
-    else
-      __process_msg "DB not running as a container"
-    fi
+  local db_container_name="db"
+  local db_container=$(sudo docker ps -a --format "{{.Names}}" | grep -w "db") || true
+  if [ ! -z "$db_container" ]; then
+    __process_msg "Found a stopped $db_container_name container, starting it"
+    sudo docker start "$db_container_name"
+    sleep 3
+  else
+    __process_msg "DB not running as a container"
   fi
 }
 
 __restart_admiral() {
-  if $DOCKER_UPGRADE_REQUIRED; then
-    __process_msg "Booting admiral container"
-    source $SCRIPTS_DIR/$ARCHITECTURE/$OPERATING_SYSTEM/boot_admiral.sh
-    sleep 5
-  fi
+  __process_msg "Booting admiral container"
+  source $SCRIPTS_DIR/$ARCHITECTURE/$OPERATING_SYSTEM/boot_admiral.sh
+  sleep 5
 }
 
 __restart() {
-  if $DOCKER_UPGRADE_REQUIRED; then
-    IS_RESTART=true
-    export skip_starting_services=true
-    source $SCRIPTS_DIR/restart.sh
-    IS_RESTART=false
-  fi
+  IS_RESTART=true
```

Comment:
should this be done in every `upgrade` flow?

Reply:
resetting docker proxy envs will restart docker anyway, which means its better to just restart services cleanly rather than letting docker handle them with defaults